addon/authenticators/oauth2-password-grant.js

```diff
@@ -320,15 +320,33 @@ export default BaseAuthenticator.extend({
     if (!isEmpty(clientIdHeader)) {
       merge(options.headers, clientIdHeader);
     }
+
+    let rejectWithResponse = this.get('rejectWithResponse');
     return new RSVP.Promise((resolve, reject) => {
       fetch(url, options).then((response) => {
         response.text().then((text) => {
-          let json = text ? JSON.parse(text) : {};
-          if (!response.ok) {
-            response.responseJSON = json;
+          let json;
+          if (rejectWithResponse && !response.ok) {
```

Comment:
Is `rejectWithResponse` actually relevant here? I think this always rejects with the response and `authenticate` rejects with either the response or the body? I think if this check is removed from here it should also be possible to make this a little less repetitive?